packages/amplify-codegen/src/commands/generateStatementsAndType.js

```diff
@@ -47,9 +47,11 @@ async function generateStatementsAndTypes(context, forceDownloadSchema, maxDepth
     ({ projectPath } = context.amplify.getEnvInfo());
   }
 
-  await context.amplify.executeProviderUtils(context, 'awscloudformation', 'compileSchema', {
-    forceCompile: true,
-  });
+  if (!project.schema.endsWith('.json') && fs.existsSync(path.join(projectPath, project.schema))){
```

Comment:
I would prefer the check rely on existence of a API resource along with looking at the file extension. There is a possibility that the API is removed, but the `.graphql.yml` file is not updated. You could follow something similar to what we do in mock. Also move the check to an helper method instead of having the same check in 3 different places

https://github.com/aws-amplify/amplify-cli/blob/eee972f44363a6fb518bcda45ffbed940144853d/packages/amplify-util-mock/src/api/api.ts#L237-L253

Reply:
Fair enough. I will add this part to the check